mongo/dataset_store.go

```diff
@@ -221,28 +261,34 @@ func (m *Mongo) GetNextVersion(datasetID, edition string) (int, error) {
 }
 
 // GetVersions retrieves all version documents for a dataset edition
-func (m *Mongo) GetVersions(ctx context.Context, id, editionID, state string) (*models.VersionResults, error) {
+func (m *Mongo) GetVersions(ctx context.Context, id, editionID, state string, offset, limit int) (*models.VersionResults, error) {
 	s := m.Session.Copy()
 	defer s.Close()
 
+	var q *mgo.Query
+
 	selector := buildVersionsQuery(id, editionID, state)
 
-	iter := s.DB(m.Database).C("instances").Find(selector).Iter()
+	q = s.DB(m.Database).C("instances").Find(selector)
+	totalCount, err := q.Count()
+	if err != nil {
+		log.Event(ctx, "error counting items", log.ERROR, log.Error(err))
+	}
+	log.Event(ctx, "totalcount", log.Data{"total_count": totalCount})
```

Comment:
Do you need this log line?

If you do, think about adding the log level, e.g. log.INFO